.github/workflows/github-metadata.yml

```diff
@@ -0,0 +1,23 @@
+name: github_metadata
+on:
+  pull_request_target:
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+
+jobs:
+  check:
+    name: protect github metadata
+    if: ${{ !contains(github.event.pull_request.labels.*.name, 'github-metadata-change') }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 1
+    env:
+      GIT_DIR: git-dir.git
```

Comment:
What does this do?

Reply:
This makes it so the `git init` below knows where to write to.  It's basically an alternative to separately passing `--git-dir` to every single git command.